setup.cfg

```diff
@@ -1,59 +1,102 @@
 test =
     pytest
     pytest-astropy
     pytest-mpl>=0.12
-    sphinx
 docs =
+    pytest-doctestplus>=0.9.0  # First version with the sphinx ext
     sphinx
     sphinx-automodapi
-    sunpy-sphinx-theme
-    pytest-doctestplus>=0.9.0  # First version with the sphinx ext
     sphinx-changelog>=1.1.0
+    sunpy-sphinx-theme
 animate =
-    sunpy>=2.0rc1
+    sunpy>=3.0.0
+
+[options.packages.find]
+exclude = ndcube._dev
 
 [tool:pytest]
 testpaths = "ndcube" "docs"
-norecursedirs = ".tox" "build" "docs[\/]_build" "docs[\/]generated" "*.egg-info" "astropy_helpers" "examples"
+norecursedirs = ".tox" "build" "docs[\/]_build" "docs[\/]generated" "*.egg-info" "examples"
 doctest_plus = enabled
 doctest_optionflags = NORMALIZE_WHITESPACE FLOAT_CMP ELLIPSIS
 text_file_format = rst
 mpl-results-path = figure_test_images
 mpl-use-full-test-name = True
-addopts = --doctest-rst
+addopts = -m "not mpl_image_compare" --doctest-rst --doctest-ignore-import-errors -p no:unraisableexception -p no:threadexception
 markers =
     remote_data: marks this test function as needing remote data.
     online: marks this test function as needing online connectivity.
-    array_compare
-    mpl_image_compare
 remote_data_strict = True
 filterwarnings =
-    ignore
-    always::DeprecationWarning
-    error::astropy.utils.exceptions.AstropyDeprecationWarning
+    error
     error::sunpy.util.exceptions.SunpyDeprecationWarning
+    # Do not fail on pytest config issues (i.e. missing plugins) but do show them
+    always::pytest.PytestConfigWarning
+    #
+    # A list of warnings to ignore follows. If you add to this list, you MUST
+    # add a comment or ideally a link to an issue that explains why the warning
+    # is being ignored
+    #
+    #
+    # This is due to dependencies building with a numpy version different from
+    # the local installed numpy version, but should be fine
+    # See https://github.com/numpy/numpy/issues/15748#issuecomment-598584838
+    ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:numpy.ndarray size changed:RuntimeWarning
+    # See https://github.com/astropy/astropy/pull/10570
+    ignore:leap-second auto-update failed
+    # Figure tests use agg backend so show doesn't work but we use it in peek
+    ignore:Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
+    # Recent change to astropy has made our tests emit this warning.
+    ignore:'datfix' made the change:astropy.wcs.wcs.FITSFixedWarning
+    ignore:'unitfix' made the change:astropy.wcs.wcs.FITSFixedWarning
+    # TODO: This ones NEED to be fixed in future
+    # Ignore this, oh we emit so many
+    ignore::astropy.utils.exceptions.AstropyUserWarning
+    always::DeprecationWarning
+    ignore:Creating an ndarray from rag:numpy.VisibleDeprecationWarning
+    ignore:Animating a NDCube does not support transposing the array. The world axes may not display as expected because the array will not be transposed.:UserWarning
```

Comment:
These need addressing in future.

Reply:
Should open an issue for it.

Reply:
Done: https://github.com/sunpy/ndcube/issues/424